src/nnue/network.cpp

```diff
@@ -185,7 +146,7 @@ bool Network<Arch, Transformer>::save(const std::optional<std::string>& filename
         actualFilename = filename.value();
     else
     {
-        if (evalFile.current != evalFile.defaultName)
+        if (std::string(evalFile.current) != std::string(evalFile.defaultName))
```

Comment:
we define operator != on `FixedString` above, are these conversions still necessary?

Reply:
it's a bit of a mess overall, the operator != as is defined is not great either, I'd honestly rather leave it at this point and revise all std::string usage across the codebase, favoring std::string_view and explicit conversions